asserts/database.go

```diff
@@ -93,14 +94,41 @@ func (nbs nullBackstore) SequenceMemberAfter(t *AssertionType, kp []string, afte
 	return nil, &NotFoundError{Type: t}
 }
 
+// keyNotFoundError is returned when the key with a given ID cannot be found.
+type keyNotFoundError struct {
+	keyID string
+	where string
```

Comment:
the wheres end up sounding a bit strange in the end. Maybe it's better to have a simpler approach where this wraps a full message:

```
type keyNotFoundError struct {
   msg string
}
```

Reply:
+1

Reply:
This has been simplified now.